### ISSUE TITLE
Error when deploying soroban contract to testnet: error. jsonrpc error: Parse error: invalid type: string "20", expected u32 at line 1 column 120

### ISSUE THREAD

**Issue by efritze** [stellar/soroban-cli#1136]
### What version are you using?

```sh
soroban version
```
soroban 20.0.2 (514406eb5245b60513f3880201b48efdfe6f9546)
soroban-env 20.0.0 (9c5394083725bacf32a56107d7fa2adf643dc7b5)
soroban-env interface version 85899345920
stellar-xdr 20.0.0 (24dd3c1b9fada6e2b28e8456a915adc212d6c47e)
xdr curr (bb54e505f814386a3f45172e0b7e95b7badbe969)

### What did you do?

1. Followed the hello-word tutorial at https://soroban.stellar.org/docs/getting-started/hello-world
2. Used soroban-cli 20.0.2 instead of 20.0.0-rc.4.1
3. In Cargo.toml, changed soroban-sdk version from '20.0.0-rc2.2' to '20.0.0'
4. Successfully ran ```cargo test ```
5. Failed to run 

```sh 
soroban contract deploy \
  --wasm target/wasm32-unknown-unknown/release/hello_soroban.wasm \
  --source alice \
  --network testnet
```
### What did you expect to see?

Contract ID of the successfully deployed contract.

### What did you see instead?

error: jsonrpc error: Parse error: invalid type: string "20", expected u32 at line 1 column 120

**Comment by garantor** [comment]
I had a similar situation too, The build and test were successful but I got  an error when I try to deploy the contract to testnet;

`error: jsonrpc error: Parse error: invalid type: integer `23865`, expected a string at line 1 column 228`

**Comment by jmstdyQlq** [comment]
any update or solution for this?

**Comment by leighmcculloch** [comment]
This issue looks like it could be resolved in a recent version up to v20.3.1, could you update to that version and try again?
